base/src/types.rs

```diff
@@ -106,8 +106,13 @@ pub enum Type<Id, T = AstType<Id>> {
     Record {
         /// The associated types of this record type
         types: Vec<Field<Id, Alias<Id, T>>>,
+        row: T,
+    },
+    EmptyRow,
```

Comment:
Ah, makes sense that it is temporary that there were two variants with fields on them - that was what was confusing me! Not sure if there is a hole in Elm's implementation there, but I think it would make sense to stick with the more traveled route of having a separate variant for `Row -> Type`.